### PR TITLE
Support speculative execution and revision confirmation and migrate to curp

### PR DESCRIPTION
**Pull request by pzheng1025**

### CODE REVIEW

File: xline/src/server/command.rs [hunk 1]
```diff
@@ -1,66 +1,159 @@
...
-        Self { key, request }
+    pub(crate) fn new(keys: Vec<KeyRange>, request: Vec<u8>, id: ProposeId) -> Self {
+        Self { keys, request, id }
     }
 
     /*
-    /// Get key of `Command`
-    pub(crate) fn key(&self) -> &String {
-        &self.key
+    /// Get keys of `Command`
```
Comment (rogercloud) [inline xline/src/server/command.rs:138]: remove it？

File: xline/src/server/mod.rs [hunk 2]
```diff
@@ -3,5 +3,5 @@ pub(crate) mod command;
 /// Xline server
 pub(crate) mod xline_server;
 
-pub(crate) use self::command::{Command, CommandResponse};
+//pub(crate) use self::command::{Command, CommandResponse, KeyRange, SyncRequest};
```
Comment (rogercloud) [inline xline/src/server/mod.rs:6]: remove it?

File: xline/src/storage/kvstore.rs [hunk 3]
```diff
@@ -182,18 +181,30 @@ impl KvStoreInner {
...
                 };
                 let revisions = self.index.get_range(range);
                 let mut values = self.db.get_values(&revisions);
                 kvs.append(&mut values);
             }
         }
+        kvs
+    }
+
+    #[allow(clippy::pattern_type_mismatch)]
```
Comment (rogercloud) [inline xline/src/storage/kvstore.rs:196]: There's no match here, why do we allow this clippy?

File: xline/src/server/command.rs (hunk 1 above)
Comment (pzheng1025) [reply xline/src/server/command.rs]: Done

File: xline/src/storage/kvstore.rs (hunk 3 above)
Comment (pzheng1025) [reply xline/src/storage/kvstore.rs]: Removed.

File: xline/src/server/mod.rs (hunk 2 above)
Comment (pzheng1025) [reply xline/src/server/mod.rs]: Done

**Pull request merged**